staticcheck/def.bzl

```diff
@@ -184,12 +184,29 @@ def staticcheck_analyzers(analyzers, prefix_path = "@com_github_sluongng_nogo_an
         )
 
     To enable all staticcheck analyzers:
-        load("@com_github_sluongng_nogo_analyzer//staticcheck:def.bzl", "ANALYZERS")
+        load("@com_github_sluongng_nogo_analyzer//staticcheck:def.bzl", "staticcheck_analyzers", "ANALYZERS")
 
         nogo(
             name = "nogo",
-            deps = TOOLS_NOGO + staticcheck_analyzers([ANALYZERS]),
+            deps = TOOLS_NOGO + staticcheck_analyzers(ANALYZERS),
             visibility = ["//visibility:public"],
         )
+
+    Add the '-' prefix to remove selected analyzers from ANALYZERS:
+        load("@com_github_sluongng_nogo_analyzer//staticcheck:def.bzl", "staticcheck_analyzers", "ANALYZERS")
+
+        nogo(
+            name = "nogo",
+            deps = TOOLS_NOGO + staticcheck_analyzers(ANALYZERS + ["-U1000"]),
+            visibility = ["//visibility:public"],
+        )
+
     """
-    return [prefix_path + ":" + a for a in analyzers]
+    qualified_analyzers = {}
+    for analyzer in analyzers:
+        # Remove analyzers prefixed with a '-'
+        if analyzer.startswith("-"):
+            qualified_analyzers.pop(prefix_path + ":" + analyzer[1:])
+        else:
+            qualified_analyzers[prefix_path + ":" + analyzer] = ""
+    return qualified_analyzers.keys()
```

Comment:
This added 2 things:
- Deduplication of the analyzers
- Negate filtering

The deduplication was not documented in the PR but it is good to have.
So it makes sense to use a `dict + pop()` instead of using an `array + remove()` for that effect.

LGTM.